src/mgr/MgrClient.cc

```diff
@@ -466,6 +466,18 @@ int MgrClient::service_daemon_register(
   return 0;
 }
 
+int MgrClient::service_daemon_unregister()
+{
+  if (!service_daemon) {
+    return 0;
+  }
+  service_daemon = false;
+  if (cct->_conf->name.is_client() && session && session->con) {
+    _send_open();
```

Comment:
For readability, I'd suggest we rename the open message to MgrSession or similar, and put a flag in there for whether it's meant to be an open or a close -- not functionally necessary but probably going to make a lot more sense to anyone reading the logs